src/api/Metadata.js

```diff
@@ -779,6 +829,76 @@ class Metadata extends File {
         }
     }
 
+    /**
+     * API for creating metadata on file
+     *
+     * @param {BoxItem} file - File object for which we are changing the description
+     * @param {Object} template - Metadata Redesign template
+     * @param {Function} successCallback - Success callback
+     * @param {Function} errorCallback - Error callback
+     * @return {Promise}
+     */
+    async createMetadataRedesign(
+        file: BoxItem,
+        template: MetadataTemplateInstance,
+        successCallback: Function,
+        errorCallback: ElementsErrorCallback,
+    ): Promise<void> {
+        this.errorCode = ERROR_CODE_CREATE_METADATA;
+        if (!file || !template) {
+            errorCallback(getBadItemError(), this.errorCode);
+            return;
+        }
+
+        const { id, permissions, is_externally_owned }: BoxItem = file;
+
+        if (!id || !permissions) {
+            errorCallback(getBadItemError(), this.errorCode);
+            return;
+        }
+
+        const canEdit = !!permissions.can_upload;
+        const isProperties =
+            template.templateKey === METADATA_TEMPLATE_PROPERTIES && template.scope === METADATA_SCOPE_GLOBAL;
+
+        if (!canEdit || (is_externally_owned && !isProperties)) {
+            errorCallback(getBadPermissionsError(), this.errorCode);
+            return;
+        }
+        this.successCallback = successCallback;
+        this.errorCallback = errorCallback;
+
+        try {
+            const fieldsValues = template.fields.reduce((acc, obj) => {
+                let { value } = obj;
+                // API does not accept string for float type
+                if (obj.type === 'float' && value) value = parseFloat(obj.value);
+                // API does not accept empty string for enum type
+                if (obj.type === 'enum' && value && value.length === 0) value = undefined;
+                acc[obj.key] = value;
+                return acc;
+            }, {});
+
+            const metadata = await this.xhr.post({
```

Comment:
It seems like a `template` from props is actually a template instance with already updated fields. Is it necessary to retrieve potentially same data from api and parse it do desired format, while same data is already retrieved from props?

Reply:
we get `type` property from metadata api which is different from what we assign to `templateInstance` when converting it from template

Reply:
Ok, so in the end, I changed the way we handle it. You're right, we pass an almost ready object in `template`, and only the `type` differs from what we get from the API. So now we use the `template` and just override the `type` value, which means we no longer need to use `createTemplateInstance` that was used before.